onnxruntime/core/providers/cpu/tensor/tile.cc

```diff
@@ -24,20 +24,69 @@ namespace onnxruntime {
 ONNX_CPU_OPERATOR_KERNEL(
     Tile,
     6,
-    KernelDefBuilder().TypeConstraint("T", DataTypeImpl::GetTensorType<float>()),
-    Tile<float>);
+    KernelDefBuilder().TypeConstraint("T", {DataTypeImpl::GetTensorType<float>(),
+                                            DataTypeImpl::GetTensorType<double>(),
+                                            DataTypeImpl::GetTensorType<int8_t>(),
+                                            DataTypeImpl::GetTensorType<int16_t>(),
+                                            DataTypeImpl::GetTensorType<int32_t>(),
+                                            DataTypeImpl::GetTensorType<int64_t>(),
+                                            DataTypeImpl::GetTensorType<uint8_t>(),
+                                            DataTypeImpl::GetTensorType<uint16_t>(),
+                                            DataTypeImpl::GetTensorType<uint32_t>(),
+                                            DataTypeImpl::GetTensorType<uint64_t>(),
+                                            DataTypeImpl::GetTensorType<bool>()}),
+    Tile);
+
+Status TileCoreForFixedSizeTypes(const Tensor& input_tensor, Tensor& output_tensor, const int64_t* repeats, TensorAxisCounters& input_counters, const TensorPitches& output_pitches, size_t element_size) {
+  size_t dimension_count = input_tensor.Shape().NumDimensions();
+
+  const uint8_t* input = reinterpret_cast<const uint8_t*>(input_tensor.DataRaw());
+  uint8_t* output = reinterpret_cast<uint8_t*>(output_tensor.MutableDataRaw());
+
+  // some helper variables that will be used along the way
+  size_t block_size = 0;
+  int64_t num_repeats = 0;
+  const uint8_t* copy = nullptr;
 
-template <>
-Status Tile<float>::Compute(OpKernelContext* ctx) const {
+  while (input_counters) {
+    // Copy the input data over
+    block_size = input_tensor.Shape().GetDims().back() * element_size;
```

Comment:
Since input_tensor is const, you can call input_tensor.Shape().GetDims().back() only once outside the loop.

Reply:
Thanks, fixed